truffle/src/main/ruby/core/rubinius/platform/ffi.rb

```diff
@@ -0,0 +1,287 @@
+      when Symbol
+        return type_size(find_type(type))
+      when Rubinius::NativeFunction
+        return type_size(TYPE_PTR)
+      when FFI::Enum
+        return type_size(TYPE_ENUM)
+      end
+
+      raise PrimitiveFailure, "FFI.type_size primitive failed: #{type}"
+    end
+
+    def size_to_type(size)
+      if sz = TypeSizes[size]
+        return sz
+      end
+
+      # Be like C, use int as the default type size.
+      return :int
+    end
+
+    def config(name)
+      Rubinius::Config["rbx.platform.#{name}"]
+    end
+
+    def config_hash(name)
+      vals = {}
+      section = "rbx.platform.#{name}."
+      Rubinius::Config.section section do |key, value|
+        vals[key.substring(section.size, key.length)] = value
+      end
+      vals
+    end
+
+    def errno
+      FFI::Platform::POSIX.errno
+    end
+
+  end
+
+  # Converts a char
+  add_typedef TYPE_CHAR,    :char
+
+  # Converts an unsigned char
+  add_typedef TYPE_UCHAR,   :uchar
+
+  # The C++ boolean type
+  add_typedef TYPE_BOOL,    :bool
+
+  # Converts a short
+  add_typedef TYPE_SHORT,   :short
+
+  # Converts an unsigned short
+  add_typedef TYPE_USHORT,  :ushort
+
+  # Converts an int
+  add_typedef TYPE_INT,     :int
+
+  # Converts an unsigned int
+  add_typedef TYPE_UINT,    :uint
+
+  # Converts a long
+  add_typedef TYPE_LONG,    :long
+
+  # Converts an unsigned long
+  add_typedef TYPE_ULONG,   :ulong
+
+  # Converts a size_t
+  add_typedef TYPE_ULONG,   :size_t
+
+  # Converts a long long
+  add_typedef TYPE_LL,      :long_long
+
+  # Converts an unsigned long long
+  add_typedef TYPE_ULL,     :ulong_long
+
+  # Converts a float
+  add_typedef TYPE_FLOAT,   :float
+
+  # Converts a double
+  add_typedef TYPE_DOUBLE,  :double
+
+  # Converts a pointer to opaque data
+  add_typedef TYPE_PTR,     :pointer
+
+  # For when a function has no return value
+  add_typedef TYPE_VOID,    :void
+
+  # Converts NULL-terminated C strings
+  add_typedef TYPE_STRING,  :string
+
+  # Use strptr when you need to free the result of some operation.
+  add_typedef TYPE_STRPTR,  :strptr
+  add_typedef TYPE_STRPTR,  :string_and_pointer
+
+  # Use for a C struct with a char [] embedded inside.
+  add_typedef TYPE_CHARARR, :char_array
+
+  # A set of unambigious integer types
+  add_typedef TYPE_CHAR,   :int8
+  add_typedef TYPE_UCHAR,  :uint8
+  add_typedef TYPE_SHORT,  :int16
+  add_typedef TYPE_USHORT, :uint16
+  add_typedef TYPE_INT,    :int32
+  add_typedef TYPE_UINT,   :uint32
+
+  # Converts a varargs argument
+  add_typedef TYPE_VARARGS, :varargs
+
+  if Rubinius::L64
+    add_typedef TYPE_LONG,  :int64
+    add_typedef TYPE_ULONG, :uint64
+  else
+    add_typedef TYPE_LL,    :int64
+    add_typedef TYPE_ULL,   :uint64
+  end
+
+  TypeSizes = Rubinius::LookupTable.new
+  TypeSizes[1] = :char
+  TypeSizes[2] = :short
+  TypeSizes[4] = :int
+  TypeSizes[8] = Rubinius::L64 ? :long : :long_long
+
+  # Load all the platform dependent types
+
+  # MODIFIED 
```

Comment:
Yes sorry I must have been in the middle of a thought here ands stopped mid-comment.